OpenTrons/opentrons-api
# fix(app): Fix Run Completion and RunSummary malfunction

# Overview
- Call `stopRun` when tapping Run Completed/Run Failed button to fix 👇
https://github.com/Opentrons/opentrons/pull/12597#pullrequestreview-1407720327
- Call reset to clone run from the `runId`
- Fix ConfirmCancelRunModal (call `dismissCurrentRun` after `stopRun`)
- Add style to protocol name to clamp a long name.


# Test Plan
- send a protocol to a robot/dev kit (if it doesn't have any protocol)
- tap `All protocols`
- run a protocol
- tap RunCompleted/RunFailed button
Open Desktop app and there is no `Go to Run` button in the robot card
- tap Run again in RunSummary
Reload Check Desktop app and there is `Go to Run` button in the robot card
- tap cancel button and tap cancel in the modal
ODD app shows RunSummary
Reload Desktop app and there is no `Go to Run` button in the robot card



# Changelog
- call dismissCurrentRun from ConfirmRunCancelModal to cancel a run properly
- call closeCurrentRun when tapping RunCompleted / RunCancel button
- activate trackProtocolEvent and switch from hard-code to const 


# Review requests



# Risk assessment
low
